linajea/tracking/solver.py

```diff
@@ -104,63 +117,134 @@ def _create_indicators(self):
             self.node_continuation[node] = self.num_vars + 5
             self.num_vars += 6
 
+        if self.write_struct_svm:
+            node_selected_file.close()
+            node_appear_file.close()
+            node_disappear_file.close()
+            node_split_file.close()
+            node_child_file.close()
+            node_continuation_file.close()
+
+        if self.write_struct_svm:
+            edge_selected_file = open(f"edge_selected_b{self.block_id}", 'w')
         for edge in self.graph.edges():
+            if self.write_struct_svm:
+                edge_selected_file.write("{} {} {}\n".format(edge[0], edge[1], self.num_vars))
+
             self.edge_selected[edge] = self.num_vars
             self.num_vars += 1
 
+        if self.write_struct_svm:
+            edge_selected_file.close()
+
     def _set_objective(self):
 
         logger.debug("setting objective")
 
         objective = pylp.LinearObjective(self.num_vars)
 
         # node selection and cell cycle costs
+        if self.write_struct_svm:
+            node_selected_weight_file = open(
+                f"features_node_selected_weight_b{self.block_id}", 'w')
+            node_selected_constant_file = open(
+                f"features_node_selected_constant_b{self.block_id}", 'w')
+            node_split_weight_file = open(
+                f"features_node_split_weight_b{self.block_id}", 'w')
+            node_split_constant_file = open(
+                f"features_node_split_constant_b{self.block_id}", 'w')
+            node_child_weight_or_constant_file = open(
+                f"features_node_child_weight_or_constant_b{self.block_id}", 'w')
+            node_continuation_weight_or_constant_file = open(
+                f"features_node_continuation_weight_or_constant_b{self.block_id}", 'w')
         for node in self.graph.nodes:
             objective.set_coefficient(
                 self.node_selected[node],
-                self._node_costs(node))
+                self._node_costs(node,
```

Comment:
This fails if self.write_struct_svm is false, because the file variables are not initialized. This also comes up in _update_objective